pkg/controller/util/unifiedpushClient.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aerogear/unifiedpush-operator/pkg/unifiedpush"
+)
+
+func UnifiedpushClient(kclient client.Client, reqLogger logr.Logger) (unifiedpush.UnifiedpushClient, error) {
+	listOptions := client.MatchingLabels(map[string]string{"internal": "unifiedpush"})
+
+	var foundServices corev1.ServiceList
+	err := kclient.List(context.TODO(), listOptions, &foundServices)
+	if err != nil {
+		return unifiedpush.UnifiedpushClient{}, err
+	}
+
+	if len(foundServices.Items) == 0 {
+		return unifiedpush.UnifiedpushClient{}, errors.New("Didn't find any UPS Services")
+	}
+
+	if len(foundServices.Items) > 1 {
```

Comment:
I've created [AEROGEAR-9335](https://issues.jboss.org/browse/AEROGEAR-9335) to only support provisioning of a single UPS instance (when shouldn't provision a second one even if a second CR is created).